### PR TITLE
Modernize Library

### PR DESCRIPTION
**Pull request by patricksanders**
It seems that this repo has all but been abandoned, so I figured I would give it some love.

This is a wholesale rewrite of the MARTA API Python library. It implements two API wrapper functions, `get_buses()` and `get_trains()`. These functions take various keyword arguments to filter results and return lists of `Bus` or `Train` objects.

The design goal of these changes is to simplify API integration to allow for easy development using the MARTA API. I also wanted to lay the groundwork for extensibility and testability of the marta-python package.

### CODE REVIEW

File: tests/test_api.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import pytest
+import requests_mock
```
Comment (gwengullett) [inline tests/test_api.py:2]: This import is not set in tests_require in setup.py nor requirements.txt

Comment (patricksanders) [reply tests/test_api.py:2]: Good catch. Just switched to tox to manage testing, so all should be good now. 👍

**Comment by patricksanders** [comment]
@landonreed Any chance you could review this? I'm also willing to maintain this repo if you've got too much on your plate.

File: marta/api.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    :param route (int): route number
+    :param api_key (str): API key to override environment variable
+    :return: list of Bus objects
+    """
+    if route is not None:
+        url = '{}{}/{}?apikey={}'.format(_BASE_URL, _BUS_ROUTE_PATH, str(route), api_key)
+    else:
+        url = '{}{}?apikey={}'.format(_BASE_URL, _BUS_PATH, _API_KEY)
+
+    response = requests.get(url)
```
Comment (landonreed) [inline marta/api.py:78]: Any reason why this response isn't checked for `401` or `403` status codes (as in `get_trains`)?

File: marta/api.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    endpoint = '{}{}?apikey={}'.format(_BASE_URL, _TRAIN_PATH, api_key)
+    response = requests.get(endpoint)
+
+    if response.status_code == 401 or response.status_code == 403:
+        raise APIKeyError('Your API key seems to be invalid. Try visiting {}.'.format(endpoint))
+
+    data = loads(response.text)
+    trains = [Train(t) for t in data]
+
+    trains = [t for t in trains if
```
Comment (landonreed) [inline marta/api.py:57]: This could probably use a comment summarizing the filter behaviors.

**Review by landonreed — CHANGES_REQUESTED** [review]
Looks great, sorry for the delay there!  I just had a couple of comments.

**Comment by patricksanders** [comment]
Thanks @landonreed! I'll address your comments as soon as I get a chance.

File: README.md [hunk 4]
```diff
@@ -1,7 +1,112 @@
...
+# Install tox
+pip install tox
+# Run tests
+tox
+```
+
+## Installing
+
+```
+python setup.py install
```
Comment (landonreed) [inline README.md:17]: I'm not really a python developer (so I don't know all of the conventions), but should the README mention that you need python 3? I haven't run python code in a while and got stuck on an error importing `JSONDecodeError` when using 2.7

File: marta/api.py (hunk 2 above)
Comment (patricksanders) [reply marta/api.py:78]: AFAIK the bus endpoint doesn't actually require an API key. I added a check in case that changes.

File: README.md (hunk 4 above)
Comment (patricksanders) [reply README.md:17]: This library should be compatible with 2.7 and 3.x. `JSONDecodeError` wasn't used anyway, so I removed it and the tests pass against 2.7 and 3.7. Nice catch!

File: marta/api.py (hunk 3 above)
Comment (patricksanders) [reply marta/api.py:57]: Done. Also added a `TODO` to make that less terrible to look at.

**Review by landonreed — APPROVED** [review]

**Comment by landonreed** [comment]
Great stuff, thanks for the changes, @patricksanders!

**Pull request merged**